### PR TITLE
Switch to okhttp

### PR DESCRIPTION
**Pull request by fiacc**

### CODE REVIEW

File: android/src/main/java/ie/gov/tracing/common/Events.kt [hunk 1]
```diff
@@ -22,7 +22,7 @@ class Events {
         const val ON_STATUS_CHANGED = "onStatusChanged" // tracing api status
         const val ON_EXPOSURE = "exposure"
 
-        private const val TAG = "RNExposureNotificationService"
+        private const val TAG = "RN_ENService"
```
Comment (fiacc) [inline android/src/main/java/ie/gov/tracing/common/Events.kt]: Warn on tag being too long

File: android/src/main/java/ie/gov/tracing/network/Fetcher.kt [hunk 2]
```diff
@@ -46,8 +63,138 @@ class Fetcher {
...
+        }
+
+        fun getOkClient(pin: Boolean = true, authenticate: Boolean = true, context: Context): OkHttpClient {
+
+            val authorizationInterceptor = getAuthorizationInterceptor(context)
+            val bearerAuthenticator = getBearerAuthenticator(context)
+            val builder = OkHttpClient.Builder()
+            var usePinning = pin;
+
+            if (BuildConfig.DEBUG) {
```
Comment (fiacc) [inline android/src/main/java/ie/gov/tracing/network/Fetcher.kt]: Open to suggestions on what we want to do in debug, rn-ssl-pinning adds HttpLoggingInterceptor so you can see what's going on, setting the user agent makes it easier to distinguish between the module and the actual app on the server. Might make sense to make pinning configurable?

File: android/src/main/java/ie/gov/tracing/network/Fetcher.kt [hunk 3]
```diff
@@ -270,35 +424,84 @@ class Fetcher {
...
                 }
 
                 val metric = Metric("android", event, version, payload)
 
-                val success = post("/metrics", Gson().toJson(metric), context)
-
-                if (!success) {
-                    Events.raiseEvent(Events.ERROR, "saveMetric - failed: $event")
-                    return
+                Single.fromCallable {
```
Comment (fiacc) [inline android/src/main/java/ie/gov/tracing/network/Fetcher.kt]: Do the metric post off the main thread, there seems to be a situation where android is killing it when it sends the DAILY_TRACE metric

File: android/src/main/java/ie/gov/tracing/network/Fetcher.kt (hunk 2 above)
Comment (colmharte) [reply android/src/main/java/ie/gov/tracing/network/Fetcher.kt]: Lets add option in the config method to allow it so be disabled. We removed that DEBUG config so would 't use that

**Comment by colmharte** [comment]
LGTM

**Review by colmharte — APPROVED** [review]

**Pull request merged**